### PR TITLE
Add `yarn commit` into script instead of use git commit 

### PR DESCRIPTION
**Pull request by tianyingchun**
1. fix wrong package.jsonn `main` configuration for `eslint-config-bases`
https://github.com/belgattitude/nextjs-monorepo-example/issues/2763#issuecomment-1274311976
2. add `cz` hook git commit  using `yarn commit` instead to standarize commit data.

### CODE REVIEW

**Comment by belgattitude** [comment]
Nice I'm new to commitizen.  Can you clarify how it compares to the installed conventional commilint ? 

I'm open to do a change here. But only if we also remove the existing conventional commit (package/config/adapt husky hook on pre commit)

**Comment by tianyingchun** [comment]
1. husky focus on git hooks
2. it better combine husky `prepare-commmit-msg` hooks to standard git commit message 
3. How to get Commit to define whether it was bug fix , feature added , Major changes or Minor Patch as below picture show
image
the workflow is:

1. cz-convention-changelog adapter
2. husky and commitlint (checks for lint in your commit message)
3. prettier and lint-staged(which runs lint against staged git files)
4. commitizen
5. Save it these package to devDependency of your package.json
6. Adds config.commitizen key to the root of your package.json

as above commit, you can try
1.  git add .
2.  git commit -m="something"
3.  it will show above picture inquier box interaction for developer.
4.  pass commit message to git

**Comment by belgattitude** [comment]
Very nice !

CI fail for duplicate deps, could I ask you one last thing: run `yarn dedupe`  and commit the updated yarn.lock. https://github.com/belgattitude/nextjs-monorepo-example/actions/runs/3231571191/jobs/5293288406. 

From there I can go on with remaining steps

PS:

Some notes just to be sure you're aware (when I wrote the last command I was on my phone). The reason I ask is that I wouldn't like to have two tools doing the same thing (and possibly compete).

The current  tool used in this repo is https://github.com/conventional-changelog/commitlint and is called on husky `commit-msg`

- https://github.com/belgattitude/nextjs-monorepo-example/blob/main/commitlint.config.js
- https://github.com/belgattitude/nextjs-monorepo-example/blob/main/.husky/commit-msg

Lint-staged is run from the husky pre-commit. 

- https://github.com/belgattitude/nextjs-monorepo-example/blob/main/.husky/pre-commit

There's nothing in pre-push (on small repos it makes sense to run typecheck, but in bigger repo it's perceived as slow)

So the idea if we move to commitizen

- [ ] Remove commitlint deps
- [ ] Be sure don't prevent changesets to operate 'commit' in the release action (I guess so, but I can fix afterward)
- [ ] Be sure we're in control of tasks that will be run by the tool at each step

File: .husky/prepare-commit-msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+exec < /dev/tty && npx cz --hook || true
```
Comment (belgattitude) [inline .husky/prepare-commit-msg:4]: Does it work on all platforms ?

File: package.json [hunk 2]
```diff
@@ -92,6 +97,8 @@
     "@types/prettier": "2.7.1",
     "@types/shell-quote": "1.7.1",
     "@your-org/eslint-config-bases": "workspace:^",
+    "commitizen": "4.2.5",
+    "cz-conventional-changelog": "3.3.0",
```
Comment (belgattitude) [inline package.json:101]: Does it competes with changeset ?

File: .husky/prepare-commit-msg (hunk 1 above)
Comment (tianyingchun) [reply .husky/prepare-commit-msg:4]: i used this on mac osx, no testing on windows

**Comment by tianyingchun** [comment]
>Some notes just to be sure you're aware (when I wrote the last command I was on my phone). The reason I ask is that I wouldn't like to have two tools doing the same thing (and possibly compete).


it seems that we can remove `[commitlint](https://github.com/conventional-changelog/commitlint)` use `pre-commit-msg `hook instead?

File: package.json (hunk 2 above)
Comment (tianyingchun) [reply package.json:101]: sorry forgot generate an changeset for  package.json update BTW in my locall `main` is latest code, `yarn dedupe --check` nothing changed of `.yarn.lock`

Comment (tianyingchun) [reply package.json:101]: BTW the yarn g:changeset cause of updated root/package.json i could not choose correct package for this.

image

nothing can indicates `./root` how to do this?

Comment (belgattitude) [reply package.json:99]: My question was not clear I think :) I was confused and wrongly thought that `cs-conventional-changelog` was generating the changelogs. But it's not. Sorry.

About changesets, there's no root concept. I recommend in this to add a 'patch' to all packages (I mean it's more for keep a doc)

But you don't have too. I'll resolve the comment. Nothing to do

File: .husky/prepare-commit-msg (hunk 1 above)
Comment (belgattitude) [reply .husky/prepare-commit-msg:4]: Thx, I'll try on linux to see. Would be great if it works on Windows but not required.

**Comment by belgattitude** [comment]
> it seems that we can remove [commitlint](https://github.com/conventional-changelog/commitlint) use pre-commit-msg hook instead?

I don't know enough the difference between the two tools in term of maintenance and long term usage. But yes let's do that. 

I propose this:

- Let's merge this 'as is'... (There's some conflicts with the branch, could you update with latest main ? and run `yarn dedupe` right after)

Once done, I'll create another PR to remove commilint and find a hack for windows (or you if you want)

**Comment by tianyingchun** [comment]
> 

have udpate latest `main` and merged

**Comment by belgattitude** [comment]
Thx I'll merge tomorrow.

**Comment by belgattitude** [comment]
@tianyingchun Thanks for your P/R. I've merged it.

Unfortunately I had to revert back the commitizen changes in https://github.com/belgattitude/nextjs-monorepo-example/pull/2784.

Not saying commitizen is "bad", it's more that it needs more work to be included in this example. So feel free to open a new P/R.

**Pull request merged**